reana/reana_dev/git.py

```diff
@@ -82,6 +91,195 @@ def get_current_commit(srcdir):
+
+    parsed_current_version = semver.VersionInfo.parse(current_version)
+    next_version = ""
+    if parsed_current_version.build or part == "build":
+        next_version = parsed_current_version.bump_build()
+    elif parsed_current_version.prerelease or part == "prerelease":
+        next_version = parsed_current_version.next_version("prerelease")
+    elif parsed_current_version.patch or part == "patch":
+        next_version = parsed_current_version.next_version("patch")
+    elif parsed_current_version.minor or part == "minor":
+        next_version = parsed_current_version.next_version("minor")
+    elif parsed_current_version.major or part == "major":
+        next_version = parsed_current_version.next_version("major")
+
+    return str(next_version)
+
+
+def bump_pep440_version(current_version, part=None):
+    """Bump a PEP440 version string.
+
+    :param current_version: current version to be bumped
+    :param part: part of the PEP440 version to bump (one of: [major, minor, micro, pre]).
+    :type current_version: str
+    :type part: str
+
+    :return: String representation of the next version
+    :rtype: string
+    """
+    try:
+        version = Version(current_version)
+        next_version = ""
+        if version.pre or part == "pre":
+            next_version = Version(
+                f"{version.major}.{version.minor}.{version.micro}{version.pre[0]}{version.pre[1]+1}"
+            )
+        elif version.micro or part == "micro":
+            next_version = Version(f"{version.major}.{version.minor}.{version.micro+1}")
+        elif version.minor or part == "minor":
+            next_version = Version(f"{version.major}.{version.minor}+1.0")
+        elif version.micro or part == "micro":
+            next_version = Version(f"{version.major}+1.0.0")
+
+        return str(next_version)
+    except InvalidVersion as e:
+        # FIXME if this happens means that last tag is not semver compliant, ask for manual
+        # interaction, but what is the manual interaction?
+        click.echo(f"Current {current_version} is not a valid PEP440 version.")
+
+
+def bump_component_version(component, current_version):
+    """Bump to next component version."""
+
+    def _replace_version_string(file_=None, current_version=None, next_version=None):
+        """Replace the old version string with the new one in the specified file."""
+        run_command(
+            f"sed -i.bk 's/{current_version}/{next_version}/' {file_} && "
+            f"[ -e {file_}.bk ] && rm {file_}.bk"  # Compatibility with BSD sed
+        )
+
+    version_files = get_component_version_files(component)
+    next_version = ""
+    if version_files.get(HELM_VERSION_FILE):
+        next_version = bump_semver2_version(current_version)
+        _replace_version_string(
+            file_=version_files.get(HELM_VERSION_FILE),
+            current_version=current_version,
+            next_version=next_version,
+        )
```

Comment:
What do you think about saving the args of this function in vars and call it only once at the end?